feature_reviser/transformer/generic_transformer.py

```diff
@@ -1,14 +1,107 @@
 # -*- coding: utf-8 -*-
 
 import re
-from typing import Any, Dict, List, Tuple, Union
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
 
 import pandas as pd
-from feature_engine.dataframe_checks import check_X
+from sklearn.preprocessing import FunctionTransformer
 
 from feature_reviser.transformer.base_transformer import BaseTransformer
+from feature_reviser.utils import check_ready_to_transform
 
-# pylint: disable= missing-function-docstring, unused-argument
+# pylint: disable=missing-function-docstring, unused-argument
+
+
+class FunctionsTransformer(BaseTransformer):
```

Comment:
Is this intentionally named `FunctionsTransformer`, or should we rather stick with the same name as in scikit-learn `FunctionTransformer`?

Reply:
I picked the name `Functions` to indicate that is derives from `FunctionTransformer` but since the user adds more than one I picked the plural.